pysmt/factory.py

```diff
@@ -441,6 +441,31 @@ def get_model(self, formula, solver_name=None, logic=None):
                 retval = solver.get_model()
             return retval
 
+    def get_implicant(self, formula, solver_name=None,
+                      logic=None):
+        mgr = self.environment.formula_manager
+        if logic == AUTO_LOGIC:
+            logic = get_logic(formula, self.environment)
+
+        with self.Solver(name=solver_name, logic=logic) \
+             as solver:
+            solver.add_assertion(formula)
+            check = solver.solve()
+            if not check:
+                return None
+            else:
```

Comment:
I think it could be nice to move this function inside the Solver interface. We should avoid having too much logic in the Factory.